edxval/tests/test_api.py

```diff
@@ -687,3 +677,241 @@ def test_existing_video_in_destination_course_id(self):
+        self.assert_obj_matches_dict_for_keys(
+            encoded_video,
+            encoded_video_dict,
+            ["url", "file_size", "bitrate"]
+        )
+
+    def assert_invalid_import(self, xml, course_id=None):
+        edx_video_id = "test_edx_video_id"
+        with self.assertRaises(ValCannotCreateError):
+            api.import_from_xml(xml, edx_video_id, course_id)
+        self.assertFalse(Video.objects.filter(edx_video_id=edx_video_id).exists())
+
+    def test_new_video_full(self):
+        new_course_id = "new_course_id"
+
+        xml = self.make_import_xml(
+            video_dict=constants.VIDEO_DICT_STAR,
+            encoded_video_dicts=[constants.ENCODED_VIDEO_DICT_STAR]
+        )
+        api.import_from_xml(xml, constants.VIDEO_DICT_STAR["edx_video_id"], new_course_id)
+
+        video = Video.objects.get(edx_video_id=constants.VIDEO_DICT_STAR["edx_video_id"])
+        self.assert_video_matches_dict(video, constants.VIDEO_DICT_STAR)
+        self.assert_encoded_video_matches_dict(
+            video.encoded_videos.get(profile__profile_name=constants.PROFILE_MOBILE),
+            constants.ENCODED_VIDEO_DICT_STAR
+        )
+        video.courses.get(course_id=new_course_id)
+
+    def test_new_video_minimal(self):
+        edx_video_id = "test_edx_video_id"
+
+        xml = self.make_import_xml(
+            video_dict={
+                "client_video_id": "dummy",
+                "duration": "0",
+            }
+        )
+        api.import_from_xml(xml, edx_video_id)
+
+        video = Video.objects.get(edx_video_id=edx_video_id)
+        self.assertFalse(video.encoded_videos.all().exists())
+        self.assertFalse(video.courses.all().exists())
+
+    def test_existing_video(self):
+        new_course_id = "new_course_id"
+
+        xml = self.make_import_xml(
+            video_dict={
+                "client_video_id": "new_client_video_id",
+                "duration": 0,
+            },
+            encoded_video_dicts=[
+                constants.ENCODED_VIDEO_DICT_FISH_DESKTOP,
+                {
+                    "url": "http://example.com/new_url",
+                    "file_size": 2733256,
+                    "bitrate": 1597804,
+                    "profile": "mobile",
+                },
+            ]
+        )
+        api.import_from_xml(xml, constants.VIDEO_DICT_FISH["edx_video_id"], new_course_id)
+
+        video = Video.objects.get(edx_video_id=constants.VIDEO_DICT_FISH["edx_video_id"])
+        self.assert_video_matches_dict(video, constants.VIDEO_DICT_FISH)
+        self.assert_encoded_video_matches_dict(
+            video.encoded_videos.get(profile__profile_name=constants.PROFILE_MOBILE),
+            constants.ENCODED_VIDEO_DICT_MOBILE
+        )
+        self.assertFalse(
+            video.encoded_videos.filter(profile__profile_name=constants.PROFILE_DESKTOP).exists()
+        )
+        self.assertFalse(video.courses.filter(course_id=new_course_id).exists())
```

Comment:
If the video already existed, for performance reasons, the `course_id` should have still been added to the list of courses.